File: KI/DynamicSessions/Sample/.agents/skills/azure-identity-defaultazurecredential/sample_codes/typescript/get-token.ts
import { DefaultAzureCredential } from "@azure/identity";

const tenantId = "...";
const scope = "https://management.azure.com/.default";

async function main(): Promise<void> {
  const credential = new DefaultAzureCredential({ tenantId });
  const token = await credential.getToken(scope, { tenantId });

  console.log("tenantId:", tenantId);
  console.log("scope:", scope);
  console.log("expiresOnTimestamp:", token?.expiresOnTimestamp);
  console.log("tokenPrefix:", token?.token.slice(0, 20));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: KI/DynamicSessions/Sample/.agents/skills/azure-identity-defaultazurecredential/sample_codes/python/get_token.py
from azure.identity import DefaultAzureCredential

TENANT_ID = "..."
SCOPE = "https://management.azure.com/.default"


def main() -> None:
    credential = DefaultAzureCredential()
    token = credential.get_token(SCOPE, tenant_id=TENANT_ID)

    print("tenant_id:", TENANT_ID)
    print("scope:", SCOPE)
    print("expires_on:", token.expires_on)
    print("token_prefix:", token.token[:20])


if __name__ == "__main__":
    main()
